bolt/lib/Target/AArch64/AArch64MCPlusBuilder.cpp

```diff
@@ -246,27 +271,97 @@ class AArch64MCPlusBuilder : public MCPlusBuilder {
             Inst.getOpcode() == AArch64::LDRSHXpre ||
             Inst.getOpcode() == AArch64::LDRSHXroW ||
             Inst.getOpcode() == AArch64::LDRSHXroX ||
-            Inst.getOpcode() == AArch64::LDRSHXui);
+            Inst.getOpcode() == AArch64::LDRSHXui ||
+            Inst.getOpcode() == AArch64::LDURHi ||
+            Inst.getOpcode() == AArch64::LDURHHi ||
+            Inst.getOpcode() == AArch64::LDURSHWi ||
+            Inst.getOpcode() == AArch64::LDURSHXi ||
+            Inst.getOpcode() == AArch64::LDTRHi ||
+            Inst.getOpcode() == AArch64::LDTRSHWi ||
+            Inst.getOpcode() == AArch64::LDTRSHXi);
   }
 
   bool isLDRW(const MCInst &Inst) const {
```

Comment:
Can all of these be turned into a switch statement?

Reply:
I think yes

Reply:
@ayermolo done, can you look at one more time please